controllers/registeredservice_controller.go

```diff
@@ -56,7 +65,31 @@ func (r *RegisteredServiceReconciler) Reconcile(ctx context.Context, req ctrl.Re
 	log := log.FromContext(ctx)
 
 	var rs primazaiov1alpha1.RegisteredService
-	if err := r.Client.Get(ctx, req.NamespacedName, &rs); err != nil {
+	err := r.Client.Get(ctx, req.NamespacedName, &rs)
+	if err != nil && errors.IsNotFound(err) {
+		log.Info("Registered Service not found, hanlding delete event")
+		var sc primazaiov1alpha1.ServiceCatalog
+		err = r.Get(ctx, types.NamespacedName{
+			Name:      "primaza-service-catalog",
+			Namespace: req.NamespacedName.Namespace, // "primaza-system",
+		}, &sc)
+		if err != nil {
+			log.Error(err, "Error fetching Service Catalog")
+			return ctrl.Result{}, err
+		}
+
+		si := ServiceInCatalog(sc, req.Name)
+		sc.Spec.Services = append(sc.Spec.Services[:si], sc.Spec.Services[si+1:]...)
```

Comment:
if the catalog does not contain the service, the function `ServiceInCatalog` returns `-1` and `sc.Spec.Services[:-1]`  will panic.

```go
a := []string{"a", "b", "c"}
si := -1
a = append(a[:si], a[si+1:]...)
fmt.Printf("a: %v\n", a)

---

panic: runtime error: slice bounds out of range [:-1]
```

Reply:
k, let me add a check before the remove. But reason test did not catch is that there is always a catalog entry if a registered service is being deleted.